### PR TITLE
[Feature][python] Add parameter environment to task

### LINKED ISSUE #11421: [Feature][python] Add parameter environment to task
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

we do not have environment https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/guide/security.html in task definition, we should add it

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jieguangzhou**
## Purpose of the pull request

close:  #11421

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

### CODE REVIEW

**Review by SbloodyS — DISMISSED** [review]
LGTM

**Comment by jieguangzhou** [comment]
@zhongjiajie PTAL, thanks

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py [hunk 1]
```diff
@@ -145,6 +147,8 @@ def __init__(
         # move attribute code and version after _process_definition and process_definition declare
         self.code, self.version = self.gen_code_and_version()
         # Add task to process definition, maybe we could put into property process_definition latter
+        self.environment_code = self.get_env_code(environment_name)
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py]: can we change this line to `self.environment_name = environment_name` and also change function `get_env_code` to 

```py
    @property
    def environment_code(self, env_name: str) -> str:
         """Convert environment name to code."""
         if env_name is None:
             return None
         return JavaGate().query_environment_info(env_name)
```

We should keep init function as simple as possible, complex init function will take more time in class initialize and make more risk on it

**Review by zhongjiajie — COMMENTED** [review]
others LGTM

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py (hunk 1 above)
Comment (jieguangzhou) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py]: nice

**Review by zhongjiajie — APPROVED** [review]

**Comment by zhongjiajie** [comment]
Thanks

**Pull request merged**